File: api/src/main/java/net/blueshell/api/validation/date/TodayValidator.java
package net.blueshell.api.validation.date;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

import java.sql.Date;
import java.time.LocalDate;
import java.time.ZoneId;

public class TodayValidator implements ConstraintValidator<Today, LocalDate> {

    @Override
    public boolean isValid(LocalDate value, ConstraintValidatorContext context) {
        if (value == null) {
            return true; // let @NotNull handle null
        }
        return LocalDate.now().equals(value);
    }
}

File: api/src/main/kotlin/net/blueshell/api/validation/date/TodayValidator.kt
package net.blueshell.api.validation.date

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import java.time.LocalDate

class TodayValidator : ConstraintValidator<net.blueshell.api.validation.date.Today?, LocalDate?> {
    override fun isValid(value: LocalDate?, context: ConstraintValidatorContext?): Boolean {
        if (value == null) {
            return true // let @NotNull handle null
        }
        return LocalDate.now() == value
    }
}

